examples/flutter_gallery/lib/demo/material/slider_demo.dart

```diff
@@ -28,6 +28,89 @@ Path _triangle(double size, Offset thumbCenter, {bool invert = false}) {
   return thumbPath;
 }
 
+Path _leftTriangle(double size, Offset thumbCenter, {bool invert = false}) {
+  final Path thumbPath = Path();
+  final double halfSize = size / 2.0;
+  final double sign = invert ? -1.0 : 1.0;
+  thumbPath.moveTo(thumbCenter.dx - halfSize, thumbCenter.dy);
+  thumbPath.lineTo(thumbCenter.dx + halfSize, thumbCenter.dy - size * sign);
+  thumbPath.lineTo(thumbCenter.dx + halfSize, thumbCenter.dy + size * sign);
+  thumbPath.close();
+  return thumbPath;
+}
+
+Path _rightTriangle(double size, Offset thumbCenter, {bool invert = false}) {
+  final Path thumbPath = Path();
+  final double halfSize = size / 2.0;
+  final double sign = invert ? -1.0 : 1.0;
+  thumbPath.moveTo(thumbCenter.dx + halfSize, thumbCenter.dy);
+  thumbPath.lineTo(thumbCenter.dx - halfSize, thumbCenter.dy - size * sign);
+  thumbPath.lineTo(thumbCenter.dx - halfSize, thumbCenter.dy + size * sign);
+  thumbPath.close();
+  return thumbPath;
+}
+
+class _CustomRangeThumbShape extends RangeSliderThumbShape {
+  static const double _thumbSize = 4.0;
+  static const double _disabledThumbSize = 3.0;
+
+  @override
+  Size getPreferredSize(bool isEnabled, bool isDiscrete) {
+    return isEnabled ? const Size.fromRadius(_thumbSize) : const Size.fromRadius(_disabledThumbSize);
+  }
+
+  static final Animatable<double> sizeTween = Tween<double>(
+    begin: _disabledThumbSize,
+    end: _thumbSize,
+  );
+
+  @override
+  void paint(
+    PaintingContext context,
+    Offset center, {
+    @required Animation<double> activationAnimation,
+    @required Animation<double> enableAnimation,
+    bool isDiscrete = false,
+    bool isEnabled = false,
+    bool isOnTop,
+    @required SliderThemeData sliderTheme,
+    TextDirection textDirection,
+    Thumb thumb,
+  }) {
+    final Canvas canvas = context.canvas;
+    final ColorTween colorTween = ColorTween(
+      begin: sliderTheme.disabledThumbColor,
+      end: sliderTheme.thumbColor,
+    );
+
+    final double size = _thumbSize * sizeTween.evaluate(enableAnimation);
+    Path thumbPath;
+    switch (textDirection) {
```

Comment:
The double nested switch statements adds a lot of code, I wonder if it's better to simplify to `if` statements:

```
if ((direction == rtl && thumb == start) || (direction == ltr && thumb == end)) {
  thumbPath = ...
} else {
  thumbPath = ...
}
```

Reply:
It is more code, but the style guide says to prefer switch statements for enums:

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values